### PR TITLE
Properly enumerate databases so we can correctly tell if the database called "local" is present or not.

### PR DESCRIPTION
**Pull request by coreykosak**
The "local" database is not available in a sharding configuration, so we shouldn't ask for stats about it in that case.

### CODE REVIEW

**Comment by katzj** [comment]
lgtm

File: src/mongodb.c [hunk 1]
```diff
@@ -305,6 +305,45 @@ static int mg_process_database(
...
+   bson_iter_t iter;
+   const char *name;
+   char **ret = NULL;
+   int i = 0;
+   mongoc_cursor_t *cursor;
+   const bson_t *doc;
+
+   BSON_ASSERT (client);
+
+   cursor = mongoc_client_find_databases (client, error);
```
Comment (igorpeshansky) [inline src/mongodb.c]: Should we be checking the error here? I don't know much about these APIs, but it seems that the call in line 338 will overwrite the contents of error...

Comment (coreykosak) [inline src/mongodb.c]: This is code copied exactly from the library (except I removed the test for "local"). Looks like mongoc_client_find_databases doesn't even look at "error". So this is OK for now.

We should probably consider preparing a patch to the Mongo people trying to make this better.

I'm also kind of sad that it has quadratic runtime due to all the reallocs.

Comment (coreykosak) [inline src/mongodb.c]: Probably the right fix here from an error checking perspective is just

```
if (!cursor) {
  return ret;
}
```

**Pull request merged**